### PR TITLE
Add EoC effect that deal damage

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Something Storm asked
#### Describe the solution
Expose `deal_damage()` to EoC
#### Testing
I used test eoc in documentation, got 20 damage in torso, got 20 damage in arm; used test eoc on monsters, they took some damage also
#### Additional context
I tried to squash my changes via interactive rebase, i failed, computers were a mistake

### CODE REVIEW

File: src/npctalk.cpp [hunk 1]
```diff
@@ -6279,6 +6279,65 @@ talk_effect_fun_t::func f_give_equipment( const JsonObject &jo, std::string_view
     };
 }
 
+
+talk_effect_fun_t::func f_deal_damage( const JsonObject &jo, std::string_view member,
+                                       const std::string_view src, bool is_npc )
```
Comment (GuardianDll) [inline src/npctalk.cpp:6284]:
```suggestion
                                       bool is_npc )
```

Comment (GuardianDll) [inline src/npctalk.cpp:6284]:
```suggestion
talk_effect_fun_t::func f_deal_damage( const JsonObject &jo, std::string_view member, bool is_npc )
```

File: doc/EFFECT_ON_CONDITION.md [hunk 2]
```diff
@@ -2607,6 +2607,42 @@ A loop of 10 iterations.
...
+Deal damage, the same way melee attack deals damage
+
+| Syntax | Optionality | Value  | Info |
+| --- | --- | --- | --- | 
+| "u_deal_damage" / "npc_deal_damage" | **mandatory** | string or [variable object](#variable-object) | Damage type that would be dealt |
+| "amount" | optional | int or [variable object](#variable-object) | Amount of damage that would be dealt; Default 0 | 
+| "bodypart" | optional | string or [variable object](#variable-object) | Bodypart that take the damage. Reminder that only characters can have limbs. Default is RANDOM | 
+| "arpen" | optional | int or [variable object](#variable-object) | Armor penetration of attack; Default 0 |
+| "arpen_mult" | optional | int or [variable object](#variable-object) | Multiplier for armor penetration; Default 1 | 
+| "dmg_mult" | optional | int or [variable object](#variable-object) | Mltiplier for damage amount. Default 1 |
```
Comment (Night-Pryanik) [inline doc/EFFECT_ON_CONDITION.md:2621]:
```suggestion
| "dmg_mult" | optional | int or [variable object](#variable-object) | Multiplier for damage amount. Default 1 | 
```

**Comment by RedMisao** [comment]
What does "the same way melee attack deals damage" mean? The attack can be dodged? That it can get mitigated by armor?
(also ty for this)

**Comment by GuardianDll** [comment]
Can't be dodged, but can be mitigated by armor

**Pull request merged**